### PR TITLE
VS 2019 Fixes

### PR DESCRIPTION
**Pull request by ahimberg**
The project basically already works except for the unit test project had a couple errors.
But I also added a bit to opt into the newer v142 compiler tools which needed one small fix for a std exception type moving to a different header.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/Microsoft/react-native-windows/pull/2396)

### CODE REVIEW

File: RNWCPP/PropertySheets/React.Cpp.props [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
   <PropertyGroup Label="Configuration">
     <PlatformToolset>v141</PlatformToolset>
+    <PlatformToolset Condition="'$(VisualStudioVersion)' == '16.0'">v142</PlatformToolset>
```
Comment (JunielKatarn) [inline RNWCPP/PropertySheets/React.Cpp.props]: Is this wise?
Aren't we using a v141-equivalent internally?

Note, VS2019 can still use the v141 toolset.

Comment (ahimberg) [reply RNWCPP/PropertySheets/React.Cpp.props]: What do you mean by that?  yes when building with 2017  - which is what our current cloud builds do, everything is v141.  I'm assuming when we move up to VS2019 and v142 tools in devmain we'll want to match that . (though v141 & v142 outputs are supposed to have binary compatibility)

Comment (JunielKatarn) [reply RNWCPP/PropertySheets/React.Cpp.props]: Meaning, maybe we should target the same toolset version regardles of the Visual Studio IDE version installed.

Comment (ahimberg) [reply RNWCPP/PropertySheets/React.Cpp.props]: Default projects created by VS2019 have the line I am adding, if using 2019 it targets v142.

**Pull request merged**